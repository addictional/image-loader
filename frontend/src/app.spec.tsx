import React from 'react';
import App from './app';

import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import { format } from 'util';







describe('app',()=>{
  let requests;
  let xhr : sinon.SinonFakeXMLHttpRequestStatic;
  let errorMock : any;

  beforeEach(() => {
    xhr = sinon.useFakeXMLHttpRequest();
    requests = [];
    xhr.onCreate = req => requests.push(req);

    const originalConsoleError = global.console.error;
    errorMock = jest.spyOn(global.console, 'error');
    errorMock.mockImplementation((message : any, ...otherParams : any) => {
      originalConsoleError(message, ...otherParams);
      throw new Error(format(message, ...otherParams));
    });
  });
  afterEach(() => {
    xhr.restore();
    errorMock.mockRestore();
  });
  const files = [
    {
      name: 'success.png',
      toString() {
        return this.name;
      },
    },
  ];
  it('Init state',() => {
    const component = renderer.create(
      <App/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
  // it('should set background to proper color on error', () => {
  //   const app = mount(<App />);
  //   app.find('input').simulate('change',files);
  //   expect(app.find('.wrapper').getElement().props.style.background).toEqual('#ff4d4f')
  // });

  it('should reset background color after transition end', () => {
    const app = mount(<App />);
    app.find('input').simulate('change',files);
    app.find('input').simulate('transitionEnd');
    app.find('input').simulate('transitionEnd');
    expect(app.find('.wrapper').getElement().props.style.background).toEqual('#ffffff')
  });
})