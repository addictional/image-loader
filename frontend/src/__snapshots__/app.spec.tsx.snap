// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app Init state 1`] = `
<div
  className="wrapper"
  onTransitionEnd={[Function]}
  style={
    Object {
      "background": "#ffffff",
    }
  }
>
  <div
    className="content"
  >
    <span
      className=""
    >
      <div
        className="ant-upload ant-upload-select ant-upload-select-text"
      >
        <span
          className="ant-upload"
          onClick={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          <input
            accept=""
            directory={null}
            multiple={false}
            onChange={[Function]}
            onClick={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file"
            webkitdirectory={null}
          />
          <button
            className="ant-btn"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-label="upload"
              className="anticon anticon-upload"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="upload"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
            <span>
               Upload
            </span>
          </button>
        </span>
      </div>
      <div
        className="ant-upload-list ant-upload-list-text"
      />
    </span>
  </div>
</div>
`;
